client/blocks/nps-survey/index.jsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import RecommendationSelect from './recommendation-select';
+
+class NpsSurvey extends Component {
+	static propTypes = {
+		onDismissed: PropTypes.func,
+		name: PropTypes.string
+	}
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			recommendationValue: null,
+			isSubmitting: false,
+			isSubmitted: false
+		};
+		this.handleRecommendationSelectChange = this.handleRecommendationSelectChange.bind( this );
+		this.handleFinishClick = this.handleFinishClick.bind( this );
+		this.handleDismissClick = this.handleDismissClick.bind( this );
+	}
+
+	handleRecommendationSelectChange( newRecommendationValue ) {
+		this.setState( {
+			recommendationValue: newRecommendationValue
```

Comment:
I'd do the `parseInt` here as opposed to `handleChange` function in the handler since this will ultimately live in a different part of the codebase (the reducer fn) and explicit casting should probably happen there.